Casks/bonitastudiocommunity.rb

```diff
@@ -8,6 +8,14 @@
   desc "Business process automation and optimization"
   homepage "https://www.bonitasoft.com/downloads"
 
+  livecheck do
+    url :url
+    regex(%r{href=["']?[^"' >]*?/tag/[^"' >]*?(\d+(?:[.-]\d+)+)-(\w+)["' >]}i)
+    strategy :github_latest do |page, regex|
+      page.scan(regex).map { |match| "#{match[0]}-#{match[1]}" }
+    end
```

Comment:
```suggestion
    regex(%r{href=["']?[^"' >]*?/tag/[^"' >]*?(\d+(?:[.-]\d+)+(?:-\w+)?)["' >]}i)
    strategy :github_latest
```

The `strategy` block is simply reconstituting the tag format, so it's not necessary. We just need to extend the capture group to also account for the trailing text after the hyphen like `-u0` in `2022.2-u0`.

The suggestion above also tweaks the regex to make the trailing text after the hyphen optional, as there are previous versions without it like `2021.1`, etc.